tests/app/app.test.ts

```diff
@@ -201,4 +201,22 @@ describe("app", () => {
     await app.mount(fixture);
     expect(fixture.innerHTML).toBe("parent<div></div>");
   });
+
+  test("can add functions to the bdom", async () => {
+    const steps: string[] = [];
+    class SomeComponent extends Component {
+      static template = xml`<div t-on-click="() => __globals__.plop('click')" class="my-div"/>`;
+    }
+    const app = new App(SomeComponent, {
+      globalValues: {
+        plop: (string : any) => {
```

Comment:
not a fan of the API: calling `__globals__.plop()` is not natural.
1. Find a way at compile time to destructure that object on demand, a bit like helpers and bdom are.
or
2. rename every member of the globalValue to `custom[Value]`

Find a way to protect the helpers and __globals__ from user alteration: if user's code accesses via `__globals__.myfunc()` and myfunc is not an arrow function,  then myfunc could access, modify, add to the __globals__ object (via its `this`)
Also, it is obviously a side effect of that feature, but it is another door open to memory leaks and errors due to scoping things into arbitrary, unkillable closures

Reply:
What happens when:
- A "main" app (let's the one spawning the webclient) defines such globals
- a secondary app (lets say the ones spawned by the htmleditor) doesn't define such globals.
In the secondary app, it is legitimate to want to spawn the same components that the main app would.
Does the globals object gets shared, or do we expect a crash ?

Reply:
there is no secondary app, that's why we have owl roots. they all share the same config/templates/translation function/settings and now global and user directives. should be fine

Reply:
> it is another door open to memory leaks and errors due to scoping things into arbitrary, unkillable closures

I for one am just not seeing it, mind expanding on that?

Reply:
It was just a general observation. In the history of what we did in odoo, we realized progressively that when passing callbacks everywhere we take the risk to have some objects in a closure that can hold the whole OWL Tree. In this particular case, those closures would be, indeed unkillable. But I don't think they will hold much information......